signup.php

```diff
@@ -4,21 +4,46 @@
 
 include('config/init.php');
 echoHeader('Sign up');
+$errors=array();
+global $errors;
 if(isset($_REQUEST['newSignUp'])){
-    insertUser(
-        $_REQUEST['username'], $_REQUEST['password'], $_REQUEST['name'], $_REQUEST['email']
-
-    );
+    
+$username = $_POST['username'];
+$email = $_POST['email'];
+$password = $_POST['password'];
+$name = $_POST['name'];
+   
+    if(empty($username)){
+        array_push($errors, 'Username is required');
+    }
+    if(empty($password)){
+        array_push($errors, 'Password is required');
+    }
+    if(empty($name)){
+        array_push($errors,'Name is required');
+    }
+    if(empty($email)){
+        array_push($errors,'Email is required');
+    }
+if(count($errors) == 0){
+        insertUser(
+            $_REQUEST['username'], $_REQUEST['password'], $_REQUEST['name'], $_REQUEST['email']);
+        header('location:signup.php');
```

Comment:
This _almost_ never matters, but I'd always put an `exit;` after doing a redirect like this. There's some weird edge case where the redirect happens but the serve keeps running the rest of this file which can cause weird behavior